### PR TITLE
bump codeql to node 16

### PR DESCRIPTION
**Pull request by davidgamero**
codeql-action/init needs to be bumped to node 16 to unblock PRs again

https://github.blog/changelog/2023-01-18-code-scanning-codeql-action-v1-is-now-deprecated/

image

### CODE REVIEW

**Review by jaiveerk — APPROVED** [review]

**Review by OliverMKing — APPROVED** [review]

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
 
          # Initializes the CodeQL tools for scanning.
          - name: Initialize CodeQL
-           uses: github/codeql-action/init@v1
+           uses: github/codeql-action/init@v2
```
Comment (Tatsinnit) [inline .github/workflows/codeql.yml:24]: ☕️💡 suggestion if you are updating this version then let’s make it compliant with all update: https://github.com/github/codeql-action

aka checkout to be at “v3”

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -11,6 +11,7 @@ on: # rebuild any PRs and main branch changes
 
 jobs:
    build: # make sure build/ci works properly
+      name: Run Unit Tests
```
Comment (Tatsinnit) [inline .github/workflows/unit-tests.yml:14]: Same for this file for any action used which is outdated like checkout. ☕️💡

**Review by Tatsinnit — APPROVED** [review]
🎉💡☕️ added some thoughts and things I noticed, thank you for this review.

File: .github/workflows/unit-tests.yml (hunk 2 above)
Comment (davidgamero) [reply .github/workflows/unit-tests.yml:14]: good catch, ive updated it now

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (davidgamero) [reply .github/workflows/codeql.yml:24]: you're right, ive update it

**Review by Tatsinnit — APPROVED** [review]
🎉☕️

**Review by OliverMKing — APPROVED** [review]

**Pull request merged**